docs/proposal/20250224-rosa-networking.md

```diff
@@ -0,0 +1,212 @@
+   * availabilityZones: array of AZs
+
+   Example:
+   ```
+   kind: RosaNetwork
+   metadata:
+     name: rosa-network-01
+     namespace: default
+   status:
+     publicSubnets:
+     - subnet-0d9f28ba991b93514
+     - subnet-2d7f18c09f1b43512
+     - subnet-1d7e19c0af1c4c57f
+     privateSubnets:
+     - subnet-1d9f28ba992a83514
+     - subnet-2d7f58c09f1b43512
+     - subnet-7d7e19c0af1f4d57f
+     availabilityZones:
+     - us-west-2a
+     - us-west-2b
+     - us-west-2c
+   ```
+
+   All resources created in the cloudformation stack will be tracked under `status.resources` array:
+   ```
+   kind: RosaNetwork
+   metadata:
+     name: rosa-network-01
+     namespace: default
+   status:
+     resources:
+       - resource: NATGateway1
+         id:
+         status: CREATE_IN_PROGRESS
+         reason: Eventual consistency check initiated
+       - resource: VPC
+         id: vpc-055edf3ebf27f18d6 
+         status: CREATE_COMPLETE
+         reason:
+       - resource: SecurityGroup
+         id:
+         status: CREATE_IN_PROGRESS
+         reason: Resource creation Initiated
+   ```
+   and will be reflecting the the values coming from [AWS CloudFormation API](https://pkg.go.dev/github.com/aws/aws-sdk-go-v2/service/cloudformation#Client.DescribeStackEvents) (`resource`, `id`, `status` and `reason`).
+
+   `status.conditions` of the `RosaNetwork` resource will be consistent with the CAPA conventions, example of a successful network stack creation:
+   ```
+   kind: RosaNetwork
+   metadata:
+     name: rosa-network-01
+     namespace: default
+   status:
+     conditions:
+     - lastTransitionTime: "2025-03-20T14:45:26Z"
+       status: "True"
+       type: RosaNetworkReady
+   ```
+   Example of failed network stack creation:
+   ```
+   kind: RosaNetwork
+   metadata:
+     name: rosa-network-01
+     namespace: default
+   status:
+     conditions:
+     - lastTransitionTime: "2025-03-18T13:25:16Z"
+       status: "False"
+       type: RosaNetworkReady
+       severity: Error
+       reason: ReconciliationFailed
+       message: Insufficient privileges for ...
+   ```
+   Failed deletion:
+   ```
+   kind: RosaNetwork
+   metadata:
+     name: rosa-network-01
+     namespace: default
+   status:
+     conditions:
+     - lastTransitionTime: "2025-03-18T13:25:16Z"
+       status: "False"
+       type: RosaNetworkReady
+       severity: Error
+       reason: DeletionFailed
+       message: ...
+   ```
+   
+1. Modifications in the ROSA control plane CRD & reconciler so that it would be possible to reference the `RosaNetwork` resource from control plane:
+   ```
+   kind: ROSAControlPlane
+     metadata:
+       name: hcp01-control-plane
+       namespace: default
+   spec:
+     RosaNetworkRef:
+       name: hcp01-rosa-network
+   ```
+   Should the ROSA control plane CR contain reference to ROSA network, the reconciler will read the AWS region, AZ and subnet ids parameters from the ROSA network CR.
```

Comment:
Lets also mention that end user cannot set both RosaControlPlane->spec->[subnets](https://github.com/kubernetes-sigs/cluster-api-provider-aws/blob/main/controlplane/rosa/api/v1beta2/rosacontrolplane_types.go#L95) and rosaNetworkRef at same time.

Reply:
Technically, the user can specify both, but CAPA would prefer one to the other, right?

Reply:
yes, we will need to validate that in the webook raising error if user set both.

Reply:
Added to the doc.